perception/alars/auv_detector/auv_detector/alars_cnn.py

```diff
@@ -0,0 +1,254 @@
+            '/M350/gimbal_camera/cam_info',
+            self.cam_info_callback,
+            1)
+        self.cam_info_received = False
+
+        # FSM states
+        self.state = "go_initial"
+        self.middle_point = None
+        self.reached_time = None
+
+        # Timer loop at 10Hz
+        self.timer = self.create_timer(0.1, self.timer_callback)
+        self.get_logger().info('PosePublisher initialized.')
+
+        # Camera and UAV parameters
+        self.altitude = 8.0  # meters above target
+        self.fov_x = None
+        self.fov_y = None
+        self.current_x = 0.0  # UAV map frame initial position
+        self.current_y = -15.0
+
+    def euler_to_quaternion(self, roll, pitch, yaw):
+        qx = math.sin(roll/2) * math.cos(pitch/2) * math.cos(yaw/2) - math.cos(roll/2) * math.sin(pitch/2) * math.sin(yaw/2)
+        qy = math.cos(roll/2) * math.sin(pitch/2) * math.cos(yaw/2) + math.sin(roll/2) * math.cos(pitch/2) * math.sin(yaw/2)
+        qz = math.cos(roll/2) * math.cos(pitch/2) * math.sin(yaw/2) - math.sin(roll/2) * math.sin(pitch/2) * math.cos(yaw/2)
+        qw = math.cos(roll/2) * math.cos(pitch/2) * math.cos(yaw/2) + math.sin(roll/2) * math.sin(pitch/2) * math.sin(yaw/2)
+        return qx, qy, qz, qw
+
+    def cam_info_callback(self, msg: CameraInfo):
+        # Only subscribe once
+        if not self.cam_info_received:
+            fx = msg.k[0]
+            fy = msg.k[4]
+            width = msg.width
+            height = msg.height
+
+            # Compute horizontal and vertical FOV
+            self.fov_x = 2 * math.atan(width / (2 * fx))
+            self.fov_y = 2 * math.atan(height / (2 * fy))
+            self.get_logger().info(f"Camera FOV received: FOV_x={math.degrees(self.fov_x):.1f}°, FOV_y={math.degrees(self.fov_y):.1f}°")
+            self.cam_info_received = True
+
+            # Unsubscribe after receiving
+            self.destroy_subscription(self.cam_info_sub)
+
+    def middle_callback(self, msg: PointStamped):
+        self.middle_point = (msg.point.x, msg.point.y)
+
+    def publish_pose(self, x, y, z, yaw_deg=0.0):
+        msg = PoseStamped()
+        msg.header.stamp = self.get_clock().now().to_msg()
+        msg.header.frame_id = "M350/map"
+
+        msg.pose.position.x = x
+        msg.pose.position.y = y
+        msg.pose.position.z = z
+
+        roll, pitch, yaw = 0.0, 0.0, math.radians(yaw_deg)
+        qx, qy, qz, qw = self.euler_to_quaternion(roll, pitch, yaw)
+        msg.pose.orientation.x = qx
+        msg.pose.orientation.y = qy
+        msg.pose.orientation.z = qz
+        msg.pose.orientation.w = qw
+
+        self.publisher_.publish(msg)
+
+    def timer_callback(self):
+        if not self.cam_info_received:
+            # Wait until FOV is received
+            return
+
+        goal_tolerance = 0.2
+
+        if self.state == "go_initial":
+            # Record start time when first entering this state
+            if not hasattr(self, 'go_initial_start_time'):
```

Comment:
do not use hasattr. this is _incredibly opaque and a PAIN to debug_. Define the variable in init, as None if needed, and then check for None instead. Better yet, define it with type hints....